### PR TITLE
Date registry function

### PR DESCRIPTION
**Pull request by mvilks**

### CODE REVIEW

File: template/registry.go [hunk 1]
```diff
@@ -89,6 +89,7 @@ func (o Options) GetInt(name string, fallback int, validate ...Validate[int]) (i
 func NewRegistry() Registry {
 	return Registry{
 		"datetime": datetimeRegistryFunc,
+		"date":     dateRegistryFunc,
```
Comment (janishorsts) [inline template/registry.go:92]: The registry functions are sorted by name. `date` should be before `datetime`.

File: template/registry_date.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	//
+	// NOTE: The option is not part of the default registry.
+	// Implementations SHOULD avoid creating options that conflict with these, but
+	// are encouraged to track development of these options during Tech Preview.
+	TimeZone *time.Location
+	// The predefined date formatting style to use (full, long, medium, short).
+	Style string
+}
+
+// parseDateOptions parses :datetime options.
```
[the remark this answers is not in the document]
Comment (mvilks) [reply template/registry_date.go:26]: Now you know I was copy-pasting 🫣

File: template/registry_date.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+			}
+		}
+	}
+
+	return &opts, nil
+}
+
+func dateFunc(operand any, options Options, locale language.Tag) (any, error) {
+	errorf := func(format string, args ...any) (any, error) {
+		return "", fmt.Errorf("date function: "+format, args...)
```
Comment (janishorsts) [inline template/registry_date.go:62]: To align text with error messages in other registry functions

```suggestion
		return "", fmt.Errorf("exec date function: "+format, args...)
```

File: template/registry_date.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (janishorsts) [reply template/registry_date.go:26]: we all do, we all do IT 😸

**Pull request merged**